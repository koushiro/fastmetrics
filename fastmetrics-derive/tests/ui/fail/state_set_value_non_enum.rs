use fastmetrics_derive::StateSetValue;

// This should fail because StateSetValue is designed for enums
#[derive(StateSetValue)]
struct NotAnEnum {
    field: bool,
}

fn main() {}
