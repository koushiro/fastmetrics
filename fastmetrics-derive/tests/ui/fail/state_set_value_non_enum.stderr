error: #[derive(StateSetValue)] can only be used for enums
 --> tests/ui/fail/state_set_value_non_enum.rs:5:8
  |
5 | struct NotAnEnum {
  |        ^^^^^^^^^
